### PR TITLE
Add nodestatus handler to manage node readiness status

### PR DESCRIPTION
**Pull request by fiunchinho**
This sets some fields from `MachinePool`:

- `machinePool.Spec.ProviderIDList`
- `machinePool.Status.ReadyReplicas`
- `machinePool.Status.AvailableReplicas`
- `machinePool.Status.UnavailableReplicas`
- `machinePool.Status.NodeRefs`

This is based on the upstream implementation by https://github.com/kubernetes-sigs/cluster-api-provider-azure
- Replicas and ProviderIDList reconciled here https://github.com/kubernetes-sigs/cluster-api/blob/630e25128c52edc3008e6cd839c3d647fc831d70/exp/controllers/machinepool_controller_phases.go
- Other fields together with noderefs reconciled here https://github.com/kubernetes-sigs/cluster-api/blob/3af94bb0b08f5cd06758edd1ec866e3dec6e4715/exp/controllers/machinepool_controller_noderef.go

### CODE REVIEW

**Review by tuommaki — APPROVED** [review]
Mostly looks good to me; couple questions though.

File: service/controller/resource/nodestatus/create.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+	machinePool.Status.InfrastructureReady = azureMachinePool.Status.Ready
+	if !azureMachinePool.Status.Ready {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "AzureMachinePool.Status.Ready is not true yet")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	machinePool.Status.Replicas = azureMachinePool.Status.Replicas
+	if azureMachinePool.Status.Replicas == 0 {
```
[the remark this answers is not in the document]
Comment (fiunchinho) [reply service/controller/resource/nodestatus/create.go:46]: This https://github.com/giantswarm/azure-operator/pull/1010

File: service/controller/resource/nodestatus/create.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			nodeRefs = append(nodeRefs, apicorev1.ObjectReference{
+				Kind:       node.Kind,
+				APIVersion: node.APIVersion,
+				Name:       node.Name,
+				UID:        node.UID,
+			})
+		}
+	}
+
+	if len(nodeRefs) == 0 {
```
[the remark this answers is not in the document]
Comment (fiunchinho) [reply service/controller/resource/nodestatus/create.go:183]: Yes, I think so. But the operator wouldn't error, just log that it couldn't assign noderefs and move on.

File: service/controller/resource/nodestatus/create.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuommaki) [reply service/controller/resource/nodestatus/create.go:46]: :rocket:

**Pull request merged**